tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -102,21 +103,59 @@ class AppFilterImpl(
       } else {
         appNameFiltered
       }
-      appNamelogicFiltered.filter(_.appInfo.exists(_.userName.contains(userName)))
+      appNamelogicFiltered.filter(_.appInfo.appStartInfo.exists(_.userName.contains(userName)))
     } else {
       appNameFiltered
     }
 
+    val configFiltered = if (configurationNames.nonEmpty) {
+      val userNameLogicFiltered = if (appArgs.any()) {
+        apps
+      } else {
+        userNameFiltered
+      }
+
+      if (configurationNames.toString.contains(":")) { // check for specific configuration values
+        val configFilteredResult = userNameLogicFiltered.filter { x =>
+          x.appInfo.configInfo.exists { config =>
+            val individualConfigs = configurationNames.map(str => str.split(":"))
+            val validConfigs = individualConfigs.filter(vConfig => vConfig.size == 2)
+            if (validConfigs.nonEmpty) {
+              val validConfigsMap = validConfigs.map(a => a(0) -> a(1)).toMap
+              val allConfigs = config.configName // all configs from eventlog
+              validConfigsMap.filter(configArgs =>
```

Comment:
any multi-line  filter/map etc should use {} rather then ()

Reply:
this might be more efficient to get the keyset and do intersection on that first and then compare values